tools/logger/convert.c

```diff
@@ -284,6 +287,16 @@ static void process_params(struct proc_ldc_entry *pe,
 			p += 2;
 		}
 	}
+
+	/* ensure no NULLs in format strings */
+	for (j = 0; j < i; j++) {
+		if (!pe->params[j]) {
+			pe->params[j] = (uintptr_t)missing;
```

Comment:
Mixing malloced pointers with static pointers is a bit scary, but I guess subst_mask is enough to make sure free is never called for "<missing>" .

Reply:
@jsarha Just a note: the code is already doing that.

Reply:
ok, looks good then.